src/components/pc-block-field.vue

```diff
@@ -1,432 +1,275 @@
+      const sel = window.getSelection()
+      sel.removeAllRanges()
+      sel.addRange(range)
+    }
+  },
+  computed: {
+    isLocked() {
+      return this.lockedField && this.lockedField == this.fieldProp
     },
-    validationCache = {};
+    formattedVal() {
+      let result = this.fieldValue
+      const sep = ','
 
-export default {
-    template: '#pc-block-field',
-    props: [
-        'lockedField',
-        'enableEdit',
-        'isReadOnly',
-        'fieldProp',
-        'fieldValue',
-        'prefix',
-        'suffix',
-        'fieldFromBlock'
-    ],
-    data () {
-        return {
-            islockedFieldSet: (this.lockedField || '').length > 0,
-            val: parseFloat(this.fieldValue),
-            isFocused: false,
+      if (result / 1000 >= 1) {
+        const [integer, decimal] = (result + '').split('.')
+
+        result = integer.split('').reduceRight((prev, cur, i, arr) => {
+          const iFromLeft = arr.length - i
+          let resultStr = cur + prev
+
+          if (iFromLeft % 3 == 0 && iFromLeft != 0 && i != 0) {
+            resultStr = sep + resultStr
+          }
+
+          return resultStr
+        }, '')
+
+        if (decimal) {
+          result += '.' + decimal
         }
+      }
+
+      return result
     },
-    computed: {
-        isLocked () {
-            return this.lockedField && this.lockedField == this.fieldProp
-        },
-        formattedVal () {
-            let result = this.val;
-            let sep = ',';
-
-            if (result / 1000 >= 1) {
-                let [integer, decimal] = (result + '').split('.');
-
-                result = integer.split('').reduceRight((prev, cur, i, arr) => {
-                    let resultStr = cur + prev,
-                        iFromLeft = arr.length - i;
-
-                    if (iFromLeft % 3 == 0 && iFromLeft != 0 && i != 0) {
-                        resultStr = sep + resultStr;
-                    }
-
-                    return resultStr;
-                }, '')
-
-                if (decimal) {
-                    result += '.' + decimal;
-
-                }
-
-            }
-
-            return result;
-        },
-        fieldWrapperClasses () {
-            let obj = {};
-
-            obj['pc-field--read-only'] = this.isReadOnly;
-            obj['pc-field--focused'] = this.isFocused;
-            obj['pc-field-' + this.fieldProp] = true;
-
-            return obj
-        },
-        fieldFormattedStyle () {
-            let result = {};
-            if (this.isFocused) {
-                result.display = 'none';
-            }
-
-            return result
-        },
-        fieldEditableStyle () {
-            let result = {};
-            if (!this.isFocused) {
-                result.opacity = 0;
-            }
-
-            return result
-        },
-        testType () {
-            return this.$store.state.attributes.testType
-        }
+    fieldClass() {
+      if (this.fieldProp != null) {
```

Comment:
You can define the prop like this

```js
fieldProp: {
    type: String,
    default: null,
}
```

in order to avoid the non-strict comparison here